signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/view/SignInWithAppleButton.kt

```diff
@@ -63,51 +99,66 @@ class SignInWithAppleButton @JvmOverloads constructor(
         }
     }
 
-    override fun onDetachedFromWindow() {
-        super.onDetachedFromWindow()
-        service = null
-        dialog = null
+    fun configure(service: SignInWithAppleService, client: SignInWithAppleClient) {
+        this.service = service
+        this.client = client
+
+        configureFragmentIfCreated()
+    }
+
+    private fun configureFragmentIfCreated() {
+        val service = service
+        val client = client
+
+        if (service == null || client == null) {
+            return
+        }
+
+        val fragmentIfCreated = fragmentTag?.let {
+            client.getFragmentManagerForSignInWithApple().findFragmentByTag(fragmentTag)
+                    as? SignInWebViewDialogFragment
+        }
+
+        fragmentIfCreated?.configure(service, client)
     }
 
     private fun onClick() {
         val service = service
         if (service == null) {
-            Log.w("SIGN_IN_WITH_APPLE", "No service configured")
+            Log.w("SIGN_IN_WITH_APPLE", "Service is not configured")
             return
         }
 
-        val authenticationAttempt = service.buildAuthenticationAttempt()
-        val webView = buildWebView(authenticationAttempt, service.callback)
+        val client = client
+        if (client == null) {
+            Log.w("SIGN_IN_WITH_APPLE", "Client is not configured")
+            return
+        }
+
+        fragmentTag = UUID.randomUUID().toString()
+
+        val fragment = SignInWebViewDialogFragment()
+        fragment.configure(service, client)
+        fragment.beginAuthenticationAttempt()
```

Comment:
I think it might be simpler to just create and pass the authentication attempt here, and not have the fragment be aware of the service at all.